### ISSUE TITLE
basic hw in the loop - pwm self validation

### ISSUE THREAD

**Issue by rusefillc** [rusefi/rusefi#2028]
we already have hw pins on test discovery

we already have logic capture code we just need to revive if broken

we have all pieces to self check eth gppwm waste etc duth cycle at lease

this might push us to apply more settings on the fly more better which I would love to see

**Comment by rusefillc** [comment]
@mck1117  says
you can probably just hw sniff the same pin that we output to
since IDR reads the actual state of the pin even in output mode

**Comment by rusefillc** [comment]
set_trigger_simulator_pin 0 none
set_trigger_simulator_pin 1 none
set_idle_pin PD1


set_ignition_pin 1 none
set_ignition_pin 2 none
set_ignition_pin 3 none
set_trigger_input_pin 0 PC7


set_logic_input_pin 0 pc6

**Comment by rusefillc** [comment]
related to #2069

**Comment by rusefillc** [comment]
#2081

**Comment by rusefillc** [comment]
@shadowm60 can you move this ticket forward?

you have commands above ^^^
you have your new debug mode
now just need to invoke those commands from a test and validate output

See https://github.com/rusefi/rusefi/blob/master/java_console/autotest/src/com/rusefi/VssHardwareLoopTest.java as a very similar test

please download https://www.jetbrains.com/idea/download/ for java IDE https://www.jetbrains.com/idea/download/

**Comment by mck1117** [comment]
@shadowm60 do we think this is done?

**Comment by shadowm60** [comment]
I don't think so, it did not had time to talk with @rusefillc about this ticket, partially done, but i've got stuck with java, it was working, but not sure if we consider it done or is there something left to do.
